components/constellation/constellation.rs

```diff
@@ -1327,26 +1327,29 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
     }
 
     fn handle_frame_size_msg(&mut self,
-                             pipeline_id: PipelineId,
-                             size: &TypedSize2D<f32, PagePx>) {
-        let result = match self.pipelines.get_mut(&pipeline_id) {
-            Some(pipeline) => {
-                if pipeline.size != Some(*size) {
-                    pipeline.size = Some(*size);
-                    let msg = ConstellationControlMsg::Resize(pipeline_id, WindowSizeData {
-                        visible_viewport: *size,
-                        initial_viewport: *size * ScaleFactor::new(1.0),
-                        device_pixel_ratio: self.window_size.device_pixel_ratio,
-                    }, WindowSizeType::Initial);
-                    Some(pipeline.event_loop.send(msg))
-                } else {
-                    None
+                             iframe_sizes: Vec<(PipelineId, Size2D<f32>)>) {
+        for size_data in iframe_sizes {
```

Comment:
nit: for (pipeline_id, untyped_size) in frame sizes.

Also, could we send the `TypedSize2D` through IPC directly? if not, why not?